packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js

```diff
@@ -484,88 +388,59 @@ export default class TableColumnResizeEditing extends Plugin {
+			if ( !this._columnIndexMap.has( cellSlot.cell ) ) {
+				this._columnIndexMap.set( cellSlot.cell, cellSlot.column );
+			}
 
-		const tableWidthAttributeOld = modelTable.getAttribute( 'tableWidth' );
-		const tableWidthAttributeNew = viewFigure.getStyle( 'width' );
+			if ( !columnWidthsInPx[ cellSlot.column ] || domCellWidth < columnWidthsInPx[ cellSlot.column ] ) {
+				columnWidthsInPx[ cellSlot.column ] = toPrecision( domCellWidth );
+			}
+		}
 
-		const isTableWidthAttributeChanged = tableWidthAttributeOld !== tableWidthAttributeNew;
+		// Insert colgroup for the table that is resized for the first time.
+		if ( ![ ...domEventData.target.findAncestor( 'table' ).getChildren() ].find( viewCol => viewCol.is( 'element', 'colgroup' ) ) ) {
+			editingView.change( viewWriter => {
+				const colgroup = viewWriter.createContainerElement( 'colgroup' );
 
-		if ( isColumnWidthsAttributeChanged || isTableWidthAttributeChanged ) {
-			if ( this._isResizingAllowed ) {
-				// Commit all changes to the model.
-				editor.model.change( writer => {
-					if ( isColumnWidthsAttributeChanged ) {
-						writer.setAttribute( 'columnWidths', columnWidthsAttributeNew, modelTable );
-					}
+				const numberOfColumns = getNumberOfColumn( modelTable, editor );
 
-					if ( isTableWidthAttributeChanged ) {
-						writer.setAttribute( 'tableWidth', `${ toPrecision( tableWidthAttributeNew ) }%`, modelTable );
-					}
-				} );
-			} else {
-				// In read-only mode revert all changes in the editing view. The model is not touched so it does not need to be restored.
-				editingView.change( writer => {
-					if ( isColumnWidthsAttributeChanged ) {
-						const columnWidths = columnWidthsAttributeOld.split( ',' );
+				for ( let i = 0; i < numberOfColumns; i++ ) {
+					const viewColElement = viewWriter.createEmptyElement( 'col' );
+					const columnWidthInPc = `${ toPrecision( columnWidthsInPx[ i ] / sumArray( columnWidthsInPx ) * 100 ) }%`;
 
-						for ( const viewCol of viewColgroup.getChildren() ) {
-							writer.setStyle( 'width', columnWidths.shift(), viewCol );
-						}
-					}
+					viewWriter.setStyle( 'width', columnWidthInPc, viewColElement );
+					viewWriter.insert( viewWriter.createPositionAt( colgroup, 'end' ), viewColElement );
+				}
 
-					if ( isTableWidthAttributeChanged ) {
-						if ( tableWidthAttributeOld ) {
-							writer.setStyle( 'width', tableWidthAttributeOld, viewFigure );
-						} else {
-							writer.removeStyle( 'width', viewFigure );
-						}
-					}
-				} );
-			}
+				viewWriter.insert( viewWriter.createPositionAt( viewTable, 'start' ), colgroup );
+			} );
 		}
 
+		this._isResizingActive = true;
+		this._resizingData = this._getResizingData( domEventData, columnWidthsInPx );
+
 		editingView.change( writer => {
-			writer.removeClass( 'table-column-resizer__active', viewResizer );
-		} );
+			const figureInitialPcWidth = this._resizingData.widths.viewFigureWidth / this._resizingData.widths.viewFigureParentWidth;
```

Comment:
I think we should comment why we are doing such change on view only. Ofc we can't do it in model because it's a temporary change meant to be seen only by the current, local user.

Also listener could be extracted.